horizon/facts/client.py

```diff
@@ -0,0 +1,99 @@
+from dataclasses import dataclass
+from typing import Optional, Annotated
+
+from fastapi import HTTPException, Depends
+from httpx import AsyncClient, Request as HttpxRequest, Response as HttpxResponse
+from loguru import logger
+from starlette import status
+from starlette.requests import Request as FastApiRequest
+from starlette.responses import Response as FastApiResponse, StreamingResponse
+
+from config import sidecar_config
+
+_client: Optional[AsyncClient] = None
+
+
+@dataclass
+class APIKeyScope:
+    organization_id: str
+    project_id: Optional[str]
+    environment_id: Optional[str]
+
+
+class FactsClient:
+    def __init__(self):
+        self._client: Optional[AsyncClient] = None
+        self._api_key_scope: Optional[APIKeyScope] = None
+
+    @property
+    def client(self) -> AsyncClient:
+        global _client
```

Comment:
Why use global client instead of self client?

Reply:
I made it a singleton to avoid creating a new client for every request